### ISSUE TITLE
no-redeclare unexpected behavior

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#5143]
**Eslint version:** master

**Scenario 1:**
I have global defined inside config file:

Config:
```json
{
    "rules": {
        "no-redeclare": [2, {"builtinGlobals": true}]
    },
    "globals":{
        "foo": false
    }
}
```
Code:
```js
var foo = 1;
```

Output:
```sh
C:\Users\gs025879\Documents\webstrom\test\gyandeep.js
  1:5  error  "foo" is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

**Scenario 2:**
I have a global variable defined in a comment inside js file.

Config:
```json
{
    "rules": {
        "no-redeclare": [2, {"builtinGlobals": true}]
    }
}
```
Code:
```js
/* globals foo */
var foo = 1;
```

Output:
no errors

Expected:
Same output as scenario 1.

<hr>
**Scenario 3:**
This is little different than scenario 1.

Config:
```json
{
    "rules": {
        "no-redeclare": [2, {"builtinGlobals": true}]
    },
    "globals":{
        "foo": false
    }
}
```
Code:
```js
foo = 1;
```

Output:
no errors

Expected:
same output as scenario 1

**Comment by mysticatea** [comment]
Scenario 2 looks like a bug indeed.
But 3 is not a redeclaration, it's just a reassignment. So as my memory, it's warned by no-native-reassign.

**Comment by nzakas** [comment]
Agreed

**Comment by gyandeeps** [comment]
False alarm here. It works fine with rc0.
